src/app/chat/views/message_list.nim

```diff
@@ -95,12 +107,28 @@ QtObject:
     self.messageIndex.del(messageId)
     self.messageReactions.del(messageId)
     self.endRemoveRows()
+    # update indexes
+    for i in countup(0, self.messages.len - 1):
+      self.messageIndex[self.messages[i].id] = i
```

Comment:
Regarding your issue of the delete not working, it might be because this change in index is done after the signal is sent, so QT recalculates stuff before the indexes are changed.